### ISSUE TITLE
Many libraries tests are not robust against line endings

### ISSUE THREAD

**Issue by tannergooding** [dotnet/runtime#66600]
There are a number of libraries tests that fail if you have a copy of the repo that was cloned using `Unix` line endings on `Windows`:
```
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'System.CodeDom.Tests'. Please check .\runtime\artifacts\bin\System.CodeDom.Tests\Release\net7.0\testResults.xml for details! [.\runtime\src\libraries\System.CodeDom\tests\System.CodeDom.Tests.csproj]
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'System.ServiceModel.Syndication.Tests'. Please check .\runtime\artifacts\bin\System.ServiceModel.Syndication.Tests\Release\net7.0\testResults.xml for details! [.\runtime\src\libraries\System.ServiceModel.Syndication\tests\System.ServiceModel.Syndication.Tests.csproj]
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'Microsoft.Extensions.FileProviders.Physical.Tests'. Please check .\runtime\artifacts\bin\Microsoft.Extensions.FileProviders.Physical.Tests\Release\net7.0\testResults.xml for details! [.\runtime\src\libraries\Microsoft.Extensions.FileProviders.Physical\tests\Microsoft.Extensions.FileProviders.Physical.Tests.csproj]
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'Microsoft.XmlSerializer.Generator.Tests'. Please check .\runtime\artifacts\bin\Microsoft.XmlSerializer.Generator.Tests\Release\net7.0\testResults.xml for details! [.\runtime\src\libraries\Microsoft.XmlSerializer.Generator\tests\Microsoft.XmlSerializer.Generator.Tests.csproj]
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'Microsoft.Extensions.Logging.Generators.Roslyn3.11.Tests'. Please check .\runtime\artifacts\bin\Microsoft.Extensions.Logging.Generators.Roslyn3.11.Tests\Release\net7.0\testResults.xml for details! [.\runtime\src\libraries\Microsoft.Extensions.Logging.Abstractions\tests\Microsoft.Extensions.Logging.Generators.Tests\Microsoft.Extensions.Logging.Generators.Roslyn3.11.Tests.csproj]
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'Microsoft.Extensions.Logging.Generators.Roslyn4.0.Tests'. Please check .\runtime\artifacts\bin\Microsoft.Extensions.Logging.Generators.Roslyn4.0.Tests\Release\net7.0\testResults.xml for details! [.\runtime\src\libraries\Microsoft.Extensions.Logging.Abstractions\tests\Microsoft.Extensions.Logging.Generators.Tests\Microsoft.Extensions.Logging.Generators.Roslyn4.0.Tests.csproj]
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'System.Speech.Tests'. Please check .\runtime\artifacts\bin\System.Speech.Tests\Release\net7.0-windows\testResults.xml for details! [.\runtime\src\libraries\System.Speech\tests\System.Speech.Tests.csproj]
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'XsltCompiler.Tests'. Please check .\runtime\artifacts\bin\XsltCompiler.Tests\Release\net7.0\testResults.xml for details! [.\runtime\src\libraries\System.Private.Xml\tests\Xslt\XsltCompiler\XsltCompiler.Tests.csproj]
.\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'LibraryImportGenerator.Unit.Tests'. Please check .\runtime\artifacts\bin\LibraryImportGenerator.Unit.Tests\Release\net7.0\testResults.xml for details! [.\runtime\src\libraries\System.Runtime.InteropServices\tests\LibraryImportGenerator.UnitTests\LibraryImportGenerator.Unit.Tests.csproj]
```

There are many reasons why a dev might have Unix style line endings, including from WSL or simply configuring their git environment to "clone as is". We should ensure these tests are robust in the face of such setups, particularly in a world where many devs are working on and interacting with multiple possible operating systems and cross-machine setups

**Comment by tannergooding** [comment]
Sorry, posted too soon.

Repro:
1. `git clone https://github.com/dotnet/runtime`
  A. Doing this from WSL -or- in a setup where `core.autocrlf = input` should be sufficient
2. `cd runtime`
3. `.\build.cmd -subset clr+libs+libs.tests -test -c Release`

Observe that many tests fail such as:
```
      System.CodeDom.Compiler.Tests.CSharpCodeGeneratorTests.GenerateCodeFromType_Invoke_Success(e: CodeTypeDeclaration { Attributes = 20482, BaseTypes = [], Comments = [], CustomAttributes = [], EndDirectives = [], ... }, o: null, expected: "public class type {\r\n    \r\n    private fieldTy"...) [FAIL]
        System.Exception :
        Expected:
        $@"public class type {{

            private fieldType field = 1;
        }}
        "
        Actual:
        $@"public class type {{

            private fieldType field = 1;
        }}
```

or
```
System.ServiceModel.Syndication.Tests.Rss20FeedFormatterTests.Write_HasFeed_SerializesExpected [FAIL]
        System.NotSupportedException : -- Expected -
        <rss xmlns:a10="http://www.w3.org/2005/Atom" version="2.0">
  === TEST EXECUTION SUMMARY ===

     System.Composition.Tests  Total: 138, Errors: 0, Failed: 0, Skipped: 0, Time: 0.474s
            <channel>
                <title />
                <description />
            </channel>

        </rss>
        -- Actual -
        <rss xmlns:a10="http://www.w3.org/2005/Atom" version="2.0">
            <channel>
                <title />
                <description />
            </channel>
        </rss>

        Stack Trace:
          C:\Users\tagoo\source\repos\runtime\src\libraries\System.ServiceModel.Syndication\tests\Utils\CompareHelper.cs(51,0): at System.ServiceModel.Syndication.Tests.CompareHelper.AssertEqualLongString(String expected, String actual)
          C:\Users\tagoo\source\repos\runtime\src\libraries\System.ServiceModel.Syndication\tests\Utils\CompareHelper.cs(39,0): at System.ServiceModel.Syndication.Tests.CompareHelper.AssertEqualWriteOutput(String expected, Action`1 writeFunction)
          C:\Users\tagoo\source\repos\runtime\src\libraries\System.ServiceModel.Syndication\tests\System\ServiceModel\Syndication\Rss20FeedFormatterTests.cs(443,0): at System.ServiceModel.Syndication.Tests.Rss20FeedFormatterTests.Write_HasFeed_SerializesExpected()
```

etc

**Comment by tannergooding** [comment]
A few of the tests, such as `System.Speech` seem to be unrelated, but are likewise not robust against various machine setups and cause users who are trying to "get started" with the repo to have a bad experience.

**Comment by Danyy427** [comment]
I would like to help. Could anyone point me in the right direction to start this thing? I am pretty new at the repo.

**Comment by danmoseley** [comment]
Hi @Danyy427 sure, I think the steps are
1. Clone dotnet/runtime in such a way that line endings are all LF instead of CRLF. You can do that with git settings -- maybe `text eol=lf` [info](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings) or simply use WSL on Windows to clone, then move that whole tree into a folder on the Windows side of things.
2. Run the libraries unit tests in the normal way, see what fails, and fix some or all of those.
LMK if you need more help.

**Comment by Danyy427** [comment]
@danmoseley I was building and running some of the tests today and they certainly fail, but testResults.xml is not generated. I created one manually but it comes out empty after the tests are run. Here is the log:

```
  Determining projects to restore...
  All projects are up-to-date for restore.
  Microsoft.Interop.SourceGeneration -> D:\Runtime\runtime\artifacts\bin\Microsoft.Interop.SourceGeneration\Debug\netstandard2.0\Microsoft.Interop.SourceGeneration.dll
  LibraryImportGenerator -> D:\Runtime\runtime\artifacts\bin\LibraryImportGenerator\Debug\netstandard2.0\Microsoft.Interop.LibraryImportGenerator.dll
  TestUtilities -> D:\Runtime\runtime\artifacts\bin\TestUtilities\Debug\net6.0\TestUtilities.dll
  System.Runtime -> D:\Runtime\runtime\artifacts\bin\System.Runtime\ref\Debug\net7.0\System.Runtime.dll
  System.Numerics.Vectors -> D:\Runtime\runtime\artifacts\bin\System.Numerics.Vectors\ref\Debug\net7.0\System.Numerics.Vectors.dll
  System.Runtime.Extensions -> D:\Runtime\runtime\artifacts\bin\System.Runtime.Extensions\ref\Debug\net7.0\System.Runtime.Extensions.dll
  System.Runtime.Numerics -> D:\Runtime\runtime\artifacts\bin\System.Runtime.Numerics\ref\Debug\net7.0\System.Runtime.Numerics.dll
  System.Resources.Writer -> D:\Runtime\runtime\artifacts\bin\System.Resources.Writer\ref\Debug\net7.0\System.Resources.Writer.dll
  System.Collections -> D:\Runtime\runtime\artifacts\bin\System.Collections\ref\Debug\net7.0\System.Collections.dll
  System.Collections.NonGeneric -> D:\Runtime\runtime\artifacts\bin\System.Collections.NonGeneric\ref\Debug\net7.0\System.Collections.NonGeneric.dll
  System.ObjectModel -> D:\Runtime\runtime\artifacts\bin\System.ObjectModel\ref\Debug\net7.0\System.ObjectModel.dll
  System.ComponentModel -> D:\Runtime\runtime\artifacts\bin\System.ComponentModel\ref\Debug\net7.0\System.ComponentModel.dll
  System.Formats.Asn1 -> D:\Runtime\runtime\artifacts\bin\System.Formats.Asn1\ref\Debug\net7.0\System.Formats.Asn1.dll
  System.Runtime.InteropServices -> D:\Runtime\runtime\artifacts\bin\System.Runtime.InteropServices\ref\Debug\net7.0\System.Runtime.InteropServices.dll
  System.ComponentModel.Primitives -> D:\Runtime\runtime\artifacts\bin\System.ComponentModel.Primitives\ref\Debug\net7.0\System.ComponentModel.Primitives.dll
  System.Drawing.Primitives -> D:\Runtime\runtime\artifacts\bin\System.Drawing.Primitives\ref\Debug\net7.0\System.Drawing.Primitives.dll
  System.ComponentModel.TypeConverter -> D:\Runtime\runtime\artifacts\bin\System.ComponentModel.TypeConverter\ref\Debug\net7.0\System.ComponentModel.TypeConverter.dll
  System.Drawing.Common -> D:\Runtime\runtime\artifacts\bin\System.Drawing.Common\ref\Debug\net7.0\System.Drawing.Common.dll
  System.Security.SecureString -> D:\Runtime\runtime\artifacts\bin\System.Security.SecureString\ref\Debug\net7.0\System.Security.SecureString.dll
  System.Memory -> D:\Runtime\runtime\artifacts\bin\System.Memory\ref\Debug\net7.0\System.Memory.dll
  Microsoft.Win32.Primitives -> D:\Runtime\runtime\artifacts\bin\Microsoft.Win32.Primitives\ref\Debug\net7.0\Microsoft.Win32.Primitives.dll
  System.Threading -> D:\Runtime\runtime\artifacts\bin\System.Threading\ref\Debug\net7.0\System.Threading.dll
  System.Net.Primitives -> D:\Runtime\runtime\artifacts\bin\System.Net.Primitives\ref\Debug\net7.0\System.Net.Primitives.dll
  System.ComponentModel.EventBasedAsync -> D:\Runtime\runtime\artifacts\bin\System.ComponentModel.EventBasedAsync\ref\Debug\net7.0\System.ComponentModel.EventBasedAsync.dll
  System.Security.Cryptography -> D:\Runtime\runtime\artifacts\bin\System.Security.Cryptography\ref\Debug\net7.0\System.Security.Cryptography.dll
  System.Security.Cryptography.X509Certificates -> D:\Runtime\runtime\artifacts\bin\System.Security.Cryptography.X509Certificates\ref\Debug\net7.0\System.Security.Cryptography.X509Certificates.dll
  System.Windows.Extensions -> D:\Runtime\runtime\artifacts\bin\System.Windows.Extensions\ref\Debug\net7.0\System.Windows.Extensions.dll
  System.Security.Principal -> D:\Runtime\runtime\artifacts\bin\System.Security.Principal\ref\Debug\net7.0\System.Security.Principal.dll
  System.Transactions.Local -> D:\Runtime\runtime\artifacts\bin\System.Transactions.Local\ref\Debug\net7.0\System.Transactions.Local.dll
  System.Text.Encoding -> D:\Runtime\runtime\artifacts\bin\System.Text.Encoding\ref\Debug\net7.0\System.Text.Encoding.dll
  System.Collections.Specialized -> D:\Runtime\runtime\artifacts\bin\System.Collections.Specialized\ref\Debug\net7.0\System.Collections.Specialized.dll
  System.Xml.ReaderWriter -> D:\Runtime\runtime\artifacts\bin\System.Xml.ReaderWriter\ref\Debug\net7.0\System.Xml.ReaderWriter.dll
  System.Security.Cryptography.Algorithms -> D:\Runtime\runtime\artifacts\bin\System.Security.Cryptography.Algorithms\ref\Debug\net7.0\System.Security.Cryptography.Algorithms.dll
  System.Data.Common -> D:\Runtime\runtime\artifacts\bin\System.Data.Common\ref\Debug\net7.0\System.Data.Common.dll
  System.Reflection.Primitives -> D:\Runtime\runtime\artifacts\bin\System.Reflection.Primitives\ref\Debug\net7.0\System.Reflection.Primitives.dll
  System.Net.WebSockets -> D:\Runtime\runtime\artifacts\bin\System.Net.WebSockets\ref\Debug\net7.0\System.Net.WebSockets.dll
  System.Reflection.Emit.ILGeneration -> D:\Runtime\runtime\artifacts\bin\System.Reflection.Emit.ILGeneration\ref\Debug\net7.0\System.Reflection.Emit.ILGeneration.dll
  System.Security.Cryptography.Csp -> D:\Runtime\runtime\artifacts\bin\System.Security.Cryptography.Csp\ref\Debug\net7.0\System.Security.Cryptography.Csp.dll
  System.Text.RegularExpressions -> D:\Runtime\runtime\artifacts\bin\System.Text.RegularExpressions\ref\Debug\net7.0\System.Text.RegularExpressions.dll
  System.Security.Claims -> D:\Runtime\runtime\artifacts\bin\System.Security.Claims\ref\Debug\net7.0\System.Security.Claims.dll
  System.Security.Principal.Windows -> D:\Runtime\runtime\artifacts\bin\System.Security.Principal.Windows\ref\Debug\net7.0\System.Security.Principal.Windows.dll
  System.Security.AccessControl -> D:\Runtime\runtime\artifacts\bin\System.Security.AccessControl\ref\Debug\net7.0\System.Security.AccessControl.dll
  System.Security.Permissions -> D:\Runtime\runtime\artifacts\bin\System.Security.Permissions\ref\Debug\net7.0\System.Security.Permissions.dll
  System.CodeDom -> D:\Runtime\runtime\artifacts\bin\System.CodeDom\ref\Debug\net7.0\System.CodeDom.dll
  System.CodeDom -> D:\Runtime\runtime\artifacts\bin\System.CodeDom\Debug\net7.0\System.CodeDom.dll
  System.CodeDom.Tests -> D:\Runtime\runtime\artifacts\bin\System.CodeDom.Tests\Debug\net7.0\System.CodeDom.Tests.dll
  ----- start Mon 04/18/2022 18:21:35.91 ===============  To repro directly: =====================================================
  pushd D:\Runtime\runtime\artifacts\bin\System.CodeDom.Tests\Debug\net7.0\
  "D:\Runtime\runtime\artifacts\bin\testhost\net7.0-windows-Debug-x64\dotnet.exe" exec --runtimeconfig System.CodeDom.Tests.runtimeconfig.json --depsfile System.CodeDom.Tests.deps.json xunit.console.dll System.CodeDom.Tests.dll -xml testResults.xml -nologo -notrait category=OuterLoop -notrait category=failing
  popd
  ===========================================================================================================
  The system cannot find the path specified.
  ----- end Mon 04/18/2022 18:21:35.92 ----- exit code 3 ----------------------------------------------------------
D:\Runtime\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'System.CodeDom.Tests'. [D:\Runtime\runtime\src\libraries\System.CodeDom\tests\System.CodeDom.Tests.csproj]

Build FAILED.

D:\Runtime\runtime\eng\testing\tests.targets(142,5): error : One or more tests failed while running tests from 'System.CodeDom.Tests'. [D:\Runtime\runtime\src\libraries\System.CodeDom\tests\System.CodeDom.Tests.csproj]
    0 Warning(s)
    1 Error(s)
```
I am running `dotnet build /t:Test` inside of `runtime\src\libraries\System.CodeDom\tests`

**Comment by danmoseley** [comment]
@Danyy427  did you build one from the root first? You need to do this to build the runtime and other shared components. After doing that then you can use dotnet build. Check out the docs on the docs folder.

**Comment by Danyy427** [comment]
@danmoseley The first time I cloned it I built the entire repo, now I just build it with `./build.cmd -projects .\src\libraries\System.CodeDom\System.CodeDom.sln`

**Comment by Danyy427** [comment]
@danmoseley I will clone the repo again and rebuild everything and see if that works, probably messed up something.

**Comment by danmoseley** [comment]
@Danyy427 sounds good. Note instead of re-cloning you can just do `git clean -fdx`.

After I pull, I do `build clr+libs+libs.tests -rc release -lc debug` from the root (same as ``build clr+libs+libs.tests` as those are defaults) and then go to a test or src folder under libraries and either do "dotnet build" or "dotnet build /t:test". I don't use Visual Studio but the build and tests should be runnable from there if you prefer it.

If I want to build all the libraries later, I go to the root and do `build libs` or `build libs+libs.tests`.

LMK if any fixes are needed to the docs.

**Comment by Danyy427** [comment]
@danmoseley No I think it will work now, it seems I have cloned it twice on different directories and I was trying to run tests on an unbuilt repo. I got confused because I had to copy it from WSL. Thanks for the help

**Comment by Danyy427** [comment]
@danmoseley I was able to fix the tests in `System.CodeDom`. There was a method there (`LineEndingsHelper.Normalize()`) that was supposed to replace the newlines in a string if they are different from `Environment.NewLine`, it basically did the following:
```
string CompiledNewline = @"
";
return text.Replace(CompiledNewLine, Environment.NewLine);
``` 
However, this method doesn't deal with our case properly as the tests contain a lot of `Environment.NewLine`s which are `\r\n` in our case, and `CompiledNewLine` is `\n` since we copied everything from a Unix system (or WSL). As a result `\r\n` is "normalized" into `\r\r\n` (`\n` which is the CompiledNewLine is replaced with `Environment.NewLine`). That is why I first converted all of `\r\n`s into `\n` and then converted all of the newlines back into `Environment.NewLine`.

Do you think this is a viable method? Here is my commit: 
https://github.com/Danyy427/runtime/commit/d227300295cdc8b46e46fb67219f145705706ae2

**Comment by danmoseley** [comment]
@Danyy427  could you check your branch? It has more files changed then you intended.

**Comment by Danyy427** [comment]
@danmoseley yup, as I committed the repository as is, the line endings of all of the files are changed, that is why it seems like I have changed 180 files.

**Comment by danmoseley** [comment]
Gotcha. The ones marked "100755 → 100644" are removing the "execute" flag from scripts. it's not a line endings thing. I'm not sure why that's happening (how much is git, how much Windows) but we certainly won't want to commit that. You can revert them, maybe with `git checkout HEAD~ **.sh` (?) and make your branch simpler.

For `.Replace(Environment.NewLine, CompiledNewline).Replace(CompiledNewline, Environment.NewLine);` -- you want to change any `\r\n` and any `\n` into whatever Environment.NewLine is? Wouldn't that be done with `.Replace("\r\n", "\n").Replace("\n", Environment.Newline)`

**Comment by Danyy427** [comment]
@danmoseley Oh, I didn’t think they were actually doing something, I’ll revert them surely.

For replacing, yes that was my goal, and my code would do the exact same thing that you wrote in our case bur I just realized it would be very wrong if we copied the repo from windows to Unix. I’ll change that

**Comment by ViktorHofer** [comment]
@tannergooding I don't think this should fall under the area-infrastructure-libraries umbrella but area-Meta. Will change the label now, please let me know if you disagree.

**Comment by Danyy427** [comment]
@danmoseley Did I just accidentally open a PR request on this repo? or is this pull request on my own repo?

**Comment by danmoseley** [comment]
Your PR is against this repo -- see dotnet/runtime in the URL: https://github.com/dotnet/runtime/pull/68231

**Comment by Danyy427** [comment]
@danmoseley I have opened another PR for the remainder of the tests (except for System.Speech)

**Comment by Danyy427** [comment]
@danmoseley System.Speech tests seem to be unrelated with Line Endings.
```
<message><![CDATA[System.EntryPointNotFoundException : Unable to find an entry point named 'waveOutGetDevCaps' in DLL 'winmm.dll'.]]></message>
```

**Comment by danmoseley** [comment]
This Speech failure is from your own local machine? 

@dotnet/interop-contrib there is no A/W suffix inference in the new source generated interop model, right?

https://github.com/dotnet/runtime/blame/main/src/libraries/Common/src/Interop/Windows/WinMm/Interop.waveOutGetDevCaps.cs#L89

if so it looks like the above edit was wrong
```
C:\Windows\System32>dumpbin /exports C:\Windows\System32\winmm.dll | find /i "waveOutGetDevCaps"
        162   9F 0000EE40 waveOutGetDevCapsA
        163   A0 0000EE60 waveOutGetDevCapsW
```

**Comment by Danyy427** [comment]
@danmoseley the issue suggests that system.speech tests are broken with LF line endings on windows.

**Comment by danmoseley** [comment]
I don't think the Speech issue is a line ending issue

**Comment by Danyy427** [comment]
@danmoseley then all of the line endings issues are fixed once generators are merged (Except for LibraryImport which I will write a issue for).

**Comment by danmoseley** [comment]
I'm expecting the Turkish culture issue to be fixed when https://github.com/dotnet/runtime/pull/67771 is merged, or if not then soon after (we need to ingest a new MSBuild that way)

Sure, I"ll re-do this test once you are merged.

**Comment by danmoseley** [comment]
does the speech test pass if you change
`internal static partial MMSYSERR waveOutGetDevCaps(IntPtr uDeviceID, ref WAVEOUTCAPS caps, int cbwoc);` to `internal static partial MMSYSERR waveOutGetDevCapsW(IntPtr uDeviceID, ref WAVEOUTCAPS caps, int cbwoc);` (adding the W)

**Comment by Danyy427** [comment]
@danmoseley the code does not compile.
```
D:\Runtime\runtime\src\libraries\System.Speech\src\Internal\Synthesis\AudioDeviceOut.cs(314,59): error CS0117: 'Interop.WinMM' does not contain a definition for 'waveOutGetDevCaps' [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
D:\Runtime\runtime\src\libraries\System.Speech\src\Internal\Synthesis\AudioDeviceOut.cs(314,59): error CS0117: 'Interop.WinMM' does not contain a definition for 'waveOutGetDevCaps' [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
D:\Runtime\runtime\src\libraries\Common\src\Interop\Windows\WinMm\Interop.waveOutGetDevCaps.cs(24,28): error CS0169: The field 'Interop.WinMM.WAVEOUTCAPS.wChannels' is never used [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
D:\Runtime\runtime\src\libraries\Common\src\Interop\Windows\WinMm\Interop.waveOutGetDevCaps.cs(18,28): error CS0169: The field 'Interop.WinMM.WAVEOUTCAPS.wMid' is never used [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
D:\Runtime\runtime\src\libraries\Common\src\Interop\Windows\WinMm\Interop.waveOutGetDevCaps.cs(26,28): error CS0169: The field 'Interop.WinMM.WAVEOUTCAPS.dwSupport' is never used [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
D:\Runtime\runtime\src\libraries\Common\src\Interop\Windows\WinMm\Interop.waveOutGetDevCaps.cs(19,28): error CS0169: The field 'Interop.WinMM.WAVEOUTCAPS.wPid' is never used [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
D:\Runtime\runtime\src\libraries\Common\src\Interop\Windows\WinMm\Interop.waveOutGetDevCaps.cs(25,28): error CS0169: The field 'Interop.WinMM.WAVEOUTCAPS.wReserved1' is never used [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
D:\Runtime\runtime\src\libraries\Common\src\Interop\Windows\WinMm\Interop.waveOutGetDevCaps.cs(23,26): error CS0169: The field 'Interop.WinMM.WAVEOUTCAPS.dwFormats' is never used [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
D:\Runtime\runtime\src\libraries\Common\src\Interop\Windows\WinMm\Interop.waveOutGetDevCaps.cs(20,26): error CS0169: The field 'Interop.WinMM.WAVEOUTCAPS.vDriverVersion' is never used [D:\Runtime\runtime\src\libraries\System.Speech\src\System.Speech.csproj]
```

**Comment by elinor-fung** [comment]
> there is no A/W suffix inference in the new source generated interop model, right?

Yep: https://github.com/dotnet/runtime/pull/68418

I guess the `System.Speech` tests are conditioned such that they don't end up running on PR? https://github.com/dotnet/runtime/pull/67189 was nearly a month ago.

**Comment by am11** [comment]
`git` overrides local rules with the repo preferred ones defined in `.gitattributes` file. When we clone a repo, git first checks out this file, overrides user-defined rules then checkout the rest.

We are using `.gitattributes` in this repo. We explicitly set `csharp` to `auto`: https://github.com/dotnet/runtime/blob/9a0b22af8ee18c028129f0c889458a718cdee262/.gitattributes#L28 implying that everything should work with either line endings. We can also set it to `text=lf`, since even Notepad.exe on Windows 10 (since 2018 update) understands LF. (all other editors on Windows supported LF for much longer)

**Comment by Danyy427** [comment]
@am11 I think that the problem arises when we attempt to compare generated code which automatically comes with <CRLF> with some form of string. For example when we have a string like:
```
string expectedCode = 
"
public void foo()
{
    return bar();
}
";

The line endings are <LF> in Linux and that's how they stay when we move them to windows. When some code is generated that is supposed to produce the code above, newlines are all `Environment.NewLine`s (which is <CRLF> in windows) so when we compare the actual with the expected we get differences. (Or perhaps I misunderstood something)

**Comment by danmoseley** [comment]
I suggest we close this now (although it sounds like possibly an edit to .gitattribute might be helpful at some point)

@Danyy427 if you'd like you could pick up another one from the "easy" list 😸

**Comment by Danyy427** [comment]
@danmoseley I opened that issue in dotnet/roslyn-sdk, I might have messed up the issue template though :) Once that is done, I'll pick another one from the list :)

**Comment by am11** [comment]
Yup, I was only pointing to the checkout aspect. Your changes are needed regardless, @Danyy427. :)

**Comment by Danyy427** [comment]
@am11 Sorry, I thought you asked what the problem was.
